pkg/settings/recording/config.go

```diff
@@ -2,15 +2,18 @@ package recording
 
 import (
 	"flag"
+	"fmt"
 )
 
 type Config struct {
-	Enabled bool `yaml:"enabled" category:"experimental"`
+	Enabled bool   `yaml:"enabled" category:"experimental"`
+	Address string `yaml:"address" category:"experimental"`
 }
 
 const (
 	flagPrefix  = "tenant-settings.recording-rules."
 	flagEnabled = flagPrefix + "enabled"
+	flagAddress = flagPrefix + "client.address"
```

Comment:
I would put those client address flags onto where the client gets consumed: "compaction-worker". (I can't see anything else consuming a ExporterRulesClient.)

If someone enables `-compaction-worker.metrics-exporter.enabled` then and only then should be validate the address flag.